sphinxcontrib/datatemplates/directive.py

```diff
@@ -137,16 +137,18 @@ def run(self):
         app = env.app
         builder = app.builder
 
+        has_source = True
```

Comment:
Why do we need a new variable? We can already test the contents of source.

Reply:
We could, but it is a bit hackish and brittle. This is more explicit. Overloading the meaning of that variable, empty string means no file.

Reply:
Is this pull request good to go, or can I do something more?